### PR TITLE
Implement add to calendar function

### PR DESCRIPTION
**Pull request by wangmeng255**

### CODE REVIEW

File: clientapp/src/components/Card.jsx [hunk 1]
```diff
@@ -11,6 +13,24 @@ startDate: string,
 endDate: string
 */
 
+const AddToCalendarDropdown = AddToCalendarHOC(Button, CalendarModal);
```
Comment (davidmjiang) [inline clientapp/src/components/Card.jsx:16]: I'm getting ReferenceError: Cannot access 'Button' before initialization. I think we need to move the Button and CalendarModal classes above this? I'm curious why you didn't run into this.

Comment (wangmeng255) [reply clientapp/src/components/Card.jsx]: What browser you use? I only test it in Chrome. However, when I rename button to CalendarButton, it is broken in both IE and Chrome.

Comment (davidmjiang) [reply clientapp/src/components/Card.jsx]: I tested in Edge. 

I think this should fix it.

File: clientapp/src/utilities/dateTimeUtilites.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                year: 'numeric',
+                month: '2-digit',
+                day: '2-digit',
+                hour: '2-digit',
+                minute: '2-digit',
+                second: '2-digit'
+            }
+        );
+        const [{ value: month },,{ value: day },,{ value: year },, {value: hour},,{value: minute},,{value:second}] = dateTimeFormat.formatToParts(date);
+
```
Comment (davidmjiang) [inline clientapp/src/utilities/dateTimeUtilites.js:15]: i'm very confused by this syntax. do you know what this is doing?

Comment (wangmeng255) [reply clientapp/src/utilities/dateTimeUtilites.js:20]: I have updated these lines to make it clear.

**Pull request merged**